### ISSUE TITLE
Repository archive settings does not support custom storage

### ISSUE THREAD

**Issue by KoltesDigital** [go-gitea/gitea#21899]
### Description

I configured a custom `s3` storage and used it for `lfs` etc. and this is working. I discovered yesterday a new `storage.repo-archive` settings section, but now the custom storage can't be used.

app.ini extract
```ini
[lfs]
STORAGE_TYPE = s3

[storage.repo-archive]
STORAGE_TYPE = s3

[storage.s3]
STORAGE_TYPE            = minio
SERVE_DIRECT            = false
MINIO_ACCESS_KEY_ID     = ...
...
```

docker logs
```
2022/11/22 09:57:50 cmd/web.go:106:runWeb() [I] Starting Gitea on PID: 18
2022/11/22 09:57:50 cmd/web.go:157:runWeb() [I] Global init
2022/11/22 09:57:50 routers/init.go:104:GlobalInitInstalled() [I] Git Version: 2.36.2, Wire Protocol Version 2 Enabled (home: /app/gitea/data/home)
2022/11/22 09:57:50 routers/init.go:107:GlobalInitInstalled() [I] AppPath: /usr/local/bin/gitea
2022/11/22 09:57:50 routers/init.go:108:GlobalInitInstalled() [I] AppWorkPath: /app/gitea
2022/11/22 09:57:50 routers/init.go:109:GlobalInitInstalled() [I] Custom path: /data/gitea
2022/11/22 09:57:50 routers/init.go:110:GlobalInitInstalled() [I] Log path: /app/gitea/log
2022/11/22 09:57:50 routers/init.go:111:GlobalInitInstalled() [I] Configuration file: /data/gitea/conf/app.ini
2022/11/22 09:57:50 routers/init.go:112:GlobalInitInstalled() [I] Run Mode: Prod
2022/11/22 09:57:50 ...dules/setting/log.go:288:newLogService() [I] Gitea v1.17.3 built with GNU Make 4.3, go1.18.7 : bindata, timetzdata, sqlite, sqlite_unlock_notify
2022/11/22 09:57:50 ...dules/setting/log.go:335:newLogService() [I] Gitea Log Mode: Console(Console:info)
2022/11/22 09:57:50 ...dules/setting/log.go:249:generateNamedLogger() [I] Router Log: Console(console:info)
2022/11/22 09:57:50 ...les/setting/cache.go:76:newCacheService() [I] Cache Service Enabled
2022/11/22 09:57:50 ...les/setting/cache.go:91:newCacheService() [I] Last Commit Cache Service Enabled
2022/11/22 09:57:50 ...s/setting/session.go:73:newSessionService() [I] Session Service Enabled
2022/11/22 09:57:50 ...s/storage/storage.go:176:initAttachments() [I] Initialising Attachment storage with type: minio
2022/11/22 09:57:50 ...les/storage/minio.go:91:NewMinioStorage() [I] Creating Minio storage at XYZ with base path attachments/
2022/11/22 09:57:51 ...s/storage/storage.go:170:initAvatars() [I] Initialising Avatar storage with type: minio
2022/11/22 09:57:51 ...les/storage/minio.go:91:NewMinioStorage() [I] Creating Minio storage at XYZ with base path avatars/
2022/11/22 09:57:51 ...s/storage/storage.go:188:initRepoAvatars() [I] Initialising Repository Avatar storage with type: minio
2022/11/22 09:57:51 ...les/storage/minio.go:91:NewMinioStorage() [I] Creating Minio storage at XYZ with base path repo-avatars/
2022/11/22 09:57:53 ...s/storage/storage.go:182:initLFS() [I] Initialising LFS storage with type: minio
2022/11/22 09:57:53 ...les/storage/minio.go:91:NewMinioStorage() [I] Creating Minio storage at XYZ with base path lfs/
2022/11/22 09:57:53 ...s/storage/storage.go:194:initRepoArchives() [I] Initialising Repository Archive storage with type: s3
2022/11/22 09:57:53 routers/init.go:57:mustInit() [F] code.gitea.io/gitea/modules/storage.Init failed: Unsupported storage type: s3
```

### Gitea Version

v1.17.3

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

Docker image gitea/gitea:1

### Database

PostgreSQL

**Comment by lunny** [comment]
I will add some test and possible fix after #22405 merge.

**Comment by lunny** [comment]
duplicated with #20703
